metadata-ingestion/src/datahub/ingestion/source/looker/lookml_source.py

```diff
@@ -1544,119 +745,49 @@ def _load_model(self, path: str) -> LookerModel:
-        sql_table_name = self._generate_fully_qualified_name(
-            sql_table_name, connection_def
-        )
-
-        return builder.make_dataset_urn_with_platform_instance(
-            platform=connection_def.platform,
-            name=sql_table_name.lower(),
-            platform_instance=connection_def.platform_instance,
-            env=connection_def.platform_env or self.source_config.env,
-        )
-
     def _get_connection_def_based_on_connection_string(
         self, connection: str
     ) -> Optional[LookerConnectionDefinition]:
         if self.source_config.connection_to_platform_map is None:
             self.source_config.connection_to_platform_map = {}
         assert self.source_config.connection_to_platform_map is not None
+        connection_def: Optional[LookerConnectionDefinition] = None
+
         if connection in self.source_config.connection_to_platform_map:
-            return self.source_config.connection_to_platform_map[connection]
+            connection_def = self.source_config.connection_to_platform_map[connection]
         elif self.looker_client:
             try:
                 looker_connection: DBConnection = self.looker_client.connection(
                     connection
                 )
             except SDKError:
                 logger.error(
-                    f"Failed to retrieve connection {connection} from Looker. This usually happens when the credentials provided are not admin credentials."
+                    f"Failed to retrieve connection {connection} from Looker. This usually happens when the "
+                    f"credentials provided are not admin credentials."
                 )
             else:
                 try:
-                    connection_def: LookerConnectionDefinition = (
-                        LookerConnectionDefinition.from_looker_connection(
-                            looker_connection
-                        )
+                    connection_def = LookerConnectionDefinition.from_looker_connection(
+                        looker_connection
                     )
 
                     # Populate the cache (using the config map) to avoid calling looker again for this connection
                     self.source_config.connection_to_platform_map[
                         connection
                     ] = connection_def
-                    return connection_def
                 except ConfigurationError:
                     self.reporter.report_warning(
                         f"connection-{connection}",
                         "Failed to load connection from Looker",
                     )
 
-        return None
+        # set the platform_env in connectionDefinition as per description provided in
+        # LookerConnectionDefinition.platform_env
+        if connection_def:
+            if connection_def.platform_env is None:
+                connection_def.platform_env = self.source_config.env
```

Comment:
this doesn't seem right

Reply:
Pleaser read this field description [https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/looker/lookml_source.py#L178](L178) and no where we were making sure that it should happened. It was crashing the sql parser, not able to recall exact issue, but it setting env to None

Reply:
this is a hack - i've suggested a better way to do it above